packages/server/src/collect.ts

```diff
@@ -22,48 +25,68 @@ const extractClassesFromHtml = (html: string): RegExp => {
         /\\|\^|\$|\{|\}|\[|\]|\(|\)|\.|\*|\+|\?|\|/g,
         '\\$&'
       );
-      htmlClasses.push(className);
+      if (className !== '' && !ignoredClasses.includes(className)) {
+        htmlClasses.push(className);
+      }
     });
     match = regex.exec(html);
   }
 
   return new RegExp(htmlClasses.join('|'), 'gm');
 };
 
-export default function collect(html: string, css: string): CollectResult {
+export default function collect(
+  html: string,
+  css: string,
+  ignoredClasses: string[] = [],
+  excludedClasses: string[] = []
+): CollectResult {
   const animations = new Set();
   const other = postcss.root();
   const critical = postcss.root();
   const stylesheet = postcss.parse(css);
-  const htmlClassesRegExp = extractClassesFromHtml(html);
+  const htmlClassesRegExp = extractClassesFromHtml(html, ignoredClasses);
+  const excludedClassesRegExp = new RegExp(excludedClasses.join('|'), 'gm');
 
   const isCritical = (rule: ChildNode) => {
     // Only check class names selectors
     if ('selector' in rule && rule.selector.startsWith('.')) {
+      const isExcluded =
+        excludedClasses.length > 0 && excludedClassesRegExp.test(rule.selector);
+      if (isExcluded) return false;
+
       return Boolean(rule.selector.match(htmlClassesRegExp));
     }
 
     return true;
   };
 
   const handleAtRule = (rule: AtRule) => {
-    let addedToCritical = false;
+    if (rule.name === 'keyframes') {
+      return;
+    }
 
-    rule.each((childRule) => {
-      if (isCritical(childRule) && !addedToCritical) {
-        critical.append(rule.clone());
-        addedToCritical = true;
+    const criticalRule = rule.clone();
+    const otherRule = rule.clone();
+
+    let removedNodesFromOther = 0;
+    criticalRule.each((childRule: ChildNode, index: number) => {
+      if (isCritical(childRule)) {
+        otherRule.nodes[index - removedNodesFromOther]?.remove();
+        // eslint-disable-next-line no-plusplus
+        removedNodesFromOther++;
```

Comment:
I think you can avoid disabling eslint here like this:
```suggestion
        removedNodesFromOther += 1;
```